src/Markup/Avalonia.Markup.Xaml/XamlIl/Runtime/IAvaloniaXamlIlParentStackProvider.cs

```diff
@@ -6,4 +6,11 @@ public interface IAvaloniaXamlIlParentStackProvider
     {
         IEnumerable<object> Parents { get; }
     }
+
+    public interface IAvaloniaXamlIlEagerParentStackProvider : IAvaloniaXamlIlParentStackProvider
```

Comment:
TODO: we should really avoid `XamlIL` in public API when possible. Not a job for this PR, as it would end up in breaking changes anyway.